src/Assert.php

```diff
@@ -2025,7 +2025,7 @@ public static function isNonEmptyList(mixed $array, string $message = ''): array
      *
      * @psalm-assert array<string, T> $array
      *
-     * @param array<string, T> $array
+     * @param mixed|array<array-key, T> $array
```

Comment:
I think this should just be `mixed`. As in, remove the `@param` declaration because the actual type is already `mixed $array`.